sklearn/utils/validation.py

```diff
@@ -866,23 +866,12 @@ def check_array(
                     "if it contains a single sample.".format(array)
                 )
 
-        # make sure we actually converted to numeric:
         if dtype_numeric and array.dtype.kind in "OUSV":
```

Comment:
I find the `"O"` case unexpected here but it seems to actually not be reachable because of `dtype_numeric` and the original dtype is object, then the condition on line 755 kicks-in and then the convertion to float64 has already happened (e.g. on lines 837 or 839).

Reply:
Let's fix that in the follow-up PR that deals with the deprecation of the object case.

Reply:
I removed the "O" since it's unreachable code.